### PR TITLE
Profile: add method for `take_heap_snapshot` with auto path

### PR DESCRIPTION
**Pull request by IanButterworth**
Just a slightly quicker/easier method to use

```
julia> using Profile

julia> Profile.take_heap_snapshot()
"/Users/ian/Documents/70704_2079406350990202.heapsnapshot"
```

cc. @vilterp @NHDaly

### CODE REVIEW

File: stdlib/Profile/src/Profile.jl [hunk 1]
```diff
@@ -1250,6 +1252,12 @@ function take_heap_snapshot(filepath::String, all_one::Bool=false)
         take_heap_snapshot(io, all_one)
     end
 end
+function take_heap_snapshot(all_one::Bool=false)
+    f = joinpath(pwd(), "$(getpid())_$(time_ns()).heapsnapshot")
+    take_heap_snapshot(f, all_one)
+    println("Heap snapshot saved to `$f`")
+    return
```
Comment (vilterp) [inline stdlib/Profile/src/Profile.jl:1259]: Maybe it could return the path?

Comment (IanButterworth) [reply stdlib/Profile/src/Profile.jl]: yeah that's better. thanks

**Review by vilterp — APPROVED** [review]

**Review by vchuravy — APPROVED** [review]

**Review by NHDaly — APPROVED** [review]

**Comment by vchuravy** [comment]
Can you change the test to use this?

File: stdlib/Profile/test/runtests.jl [hunk 2]
```diff
@@ -272,12 +272,17 @@ end
 end
 
 @testset "HeapSnapshot" begin
-    fname = tempname()
-    run(`$(Base.julia_cmd()) --startup-file=no -e "using Profile; Profile.take_heap_snapshot($(repr(fname)))"`)
+    fname = strip(String(read(
+        `$(Base.julia_cmd()) --startup-file=no -E "using Profile; Profile.take_heap_snapshot()"`
+    )), ['\n', '\"'])
```
Comment (vtjnash) [inline stdlib/Profile/test/runtests.jl:277]: You can't just randomly hack apart a Julia expression object and expect something meaningful back. Pass the thing you want directly:

```suggestion
    fname = read(`$(Base.julia_cmd()) --startup-file=no -e "using Profile; print(Profile.take_heap_snapshot())"`, String)
```

Comment (IanButterworth) [reply stdlib/Profile/test/runtests.jl:277]: makes sense. Thanks https://github.com/JuliaLang/julia/pull/47320

**Pull request merged**